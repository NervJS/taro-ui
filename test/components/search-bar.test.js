import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtSearchBar from '../../.temp/components/search-bar/index'

describe('AtSearchBar Snap', () => {
  it('render initial AtSearchBar', () => {
    const componet = renderToString(<AtSearchBar />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props className', () => {
    const componet = renderToString(<AtSearchBar className='test' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props customStyle', () => {
    const componet = renderToString(<AtSearchBar customStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props value', () => {
    const componet = renderToString(<AtSearchBar value='value' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props placeholder', () => {
    const componet = renderToString(<AtSearchBar placeholder='placeholder' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props maxlength', () => {
    const componet = renderToString(<AtSearchBar maxlength={120} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props disabled', () => {
    const componet = renderToString(<AtSearchBar fixed />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props focus', () => {
    const componet = renderToString(<AtSearchBar focus />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props disabled', () => {
    const componet = renderToString(<AtSearchBar disabled />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props showActionButton', () => {
    const componet = renderToString(<AtSearchBar showActionButton />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props actionName', () => {
    const componet = renderToString(<AtSearchBar actionName='actionName' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSearchBar -- props inputType number', () => {
    const componet = renderToString(<AtSearchBar inputType='number' />)
    expect(componet).toMatchSnapshot()
  })
})
