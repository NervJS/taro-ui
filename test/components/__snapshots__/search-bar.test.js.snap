// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtSearchBar Snap render AtSearchBar -- props actionName 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:1;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:visible;\\" class=\\"taro-text at-search-bar__placeholder\\">搜索</span></div><input confirmType=\\"search\\" value=\\"\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" maxlength=\\"140\\" type=\\"search\\"/><div style=\\"display:none;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-171px;\\" class=\\"at-search-bar__action\\">actionName</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props className 1`] = `"<div class=\\"at-search-bar test\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:1;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:visible;\\" class=\\"taro-text at-search-bar__placeholder\\">搜索</span></div><input confirmType=\\"search\\" value=\\"\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" maxlength=\\"140\\" type=\\"search\\"/><div style=\\"display:none;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-59px;\\" class=\\"at-search-bar__action\\">搜索</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props customStyle 1`] = `"<div style=\\"color:red;\\" class=\\"at-search-bar\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:1;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:visible;\\" class=\\"taro-text at-search-bar__placeholder\\">搜索</span></div><input confirmType=\\"search\\" value=\\"\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" maxlength=\\"140\\" type=\\"search\\"/><div style=\\"display:none;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-59px;\\" class=\\"at-search-bar__action\\">搜索</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props disabled 1`] = `"<div class=\\"at-search-bar at-search-bar--fixed\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:1;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:visible;\\" class=\\"taro-text at-search-bar__placeholder\\">搜索</span></div><input confirmType=\\"search\\" value=\\"\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" maxlength=\\"140\\" type=\\"search\\"/><div style=\\"display:none;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-59px;\\" class=\\"at-search-bar__action\\">搜索</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props disabled 2`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:1;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:visible;\\" class=\\"taro-text at-search-bar__placeholder\\">搜索</span></div><input confirmType=\\"search\\" value=\\"\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" disabled maxlength=\\"140\\" type=\\"search\\"/><div style=\\"display:none;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-59px;\\" class=\\"at-search-bar__action\\">搜索</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props focus 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:0;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:visible;\\" class=\\"taro-text at-search-bar__placeholder\\"></span></div><input confirmType=\\"search\\" value=\\"\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" maxlength=\\"140\\" autofocus type=\\"search\\"/><div style=\\"display:none;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:1;margin-right:0;\\" class=\\"at-search-bar__action\\">搜索</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props inputType number 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:1;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:visible;\\" class=\\"taro-text at-search-bar__placeholder\\">搜索</span></div><input confirmType=\\"search\\" value=\\"\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" maxlength=\\"140\\" type=\\"search\\"/><div style=\\"display:none;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-59px;\\" class=\\"at-search-bar__action\\">搜索</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props maxlength 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:1;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:visible;\\" class=\\"taro-text at-search-bar__placeholder\\">搜索</span></div><input confirmType=\\"search\\" value=\\"\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" maxlength=\\"140\\" type=\\"search\\"/><div style=\\"display:none;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-59px;\\" class=\\"at-search-bar__action\\">搜索</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props placeholder 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:1;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:visible;\\" class=\\"taro-text at-search-bar__placeholder\\">placeholder</span></div><input confirmType=\\"search\\" value=\\"\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" maxlength=\\"140\\" type=\\"search\\"/><div style=\\"display:none;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-59px;\\" class=\\"at-search-bar__action\\">搜索</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props showActionButton 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:1;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:visible;\\" class=\\"taro-text at-search-bar__placeholder\\">搜索</span></div><input confirmType=\\"search\\" value=\\"\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" maxlength=\\"140\\" type=\\"search\\"/><div style=\\"display:none;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:1;margin-right:0;\\" class=\\"at-search-bar__action\\">搜索</div></div>"`;

exports[`AtSearchBar Snap render AtSearchBar -- props value 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:0;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:hidden;\\" class=\\"taro-text at-search-bar__placeholder\\">搜索</span></div><input confirmType=\\"search\\" value=\\"value\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" maxlength=\\"140\\" type=\\"search\\"/><div style=\\"display:flex;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:1;margin-right:0;\\" class=\\"at-search-bar__action\\">搜索</div></div>"`;

exports[`AtSearchBar Snap render initial AtSearchBar 1`] = `"<div class=\\"at-search-bar\\"><div class=\\"at-search-bar__input-cnt\\"><div style=\\"flex-grow:1;\\" class=\\"at-search-bar__placeholder-wrap\\"><span class=\\"taro-text at-icon at-icon-search\\"></span><span style=\\"visibility:visible;\\" class=\\"taro-text at-search-bar__placeholder\\">搜索</span></div><input confirmType=\\"search\\" value=\\"\\" maxLength=\\"140\\" class=\\"weui-input at-search-bar__input\\" maxlength=\\"140\\" type=\\"search\\"/><div style=\\"display:none;\\" class=\\"at-search-bar__clear\\"><span class=\\"taro-text at-icon at-icon-close-circle\\"></span></div></div><div style=\\"opacity:0;margin-right:-59px;\\" class=\\"at-search-bar__action\\">搜索</div></div>"`;
